cmake/package_helpers.cmake

```diff
@@ -94,10 +117,19 @@ macro(ginkgo_add_external_target new_target external_name includedir libdir buil
         set_target_properties(${new_target} PROPERTIES IMPORTED_LOCATION_RELEASE ${${external_name}_LIBRARY_RELEASE})
         set_target_properties(${new_target} PROPERTIES IMPORTED_LOCATION_DEBUG ${${external_name}_LIBRARY_DEBUG})
         # Since we do not really manage other build types, let's globally use the DEBUG symbols
-        if (NOT CMAKE_BUILD_TYPE MATCHES "[Rr][Ee][Ll][Ee][Aa][Ss][Ee]"
-            AND NOT CMAKE_BUILD_TYPE MATCHES "[Dd][Ee][Bb][Uu][Gg]")
+        if(MSVC)
+        # Only Debug build uses MDd or MTd, and others use MD or MT.
+        # MSVC would like to use same runtime library, so we use Debug third-party in Debug and Release third-party in Release.
+            set_target_properties(${new_target} PROPERTIES IMPORTED_LOCATION_DEBUG
+                    ${${external_name}_LIBRARY_DEBUG})
             set_target_properties(${new_target} PROPERTIES IMPORTED_LOCATION
-                ${${external_name}_LIBRARY_DEBUG})
+                    ${${external_name}_LIBRARY_RELEASE})
```

Comment:
Do you need to change this general `IMPORTED_LOCATION` between DEBUG and RELEASE location depending on the type considered? Or do you have to set this at all considering both IMPORTED_LOCATION_DEBUG and IMPORTED_LOCATION_RELEASE are already set?

Reply:
If `IMPORTED_LOCATION_${type}` is not set, cmake takes `IMPORTED_LOCATION`.
By doing so, it can used by `RELWITHDEBINFO` or `MINSIZEREL`

Reply:
right, that makes sense, I was simply confused with the previous line also being there